### PR TITLE
Move `SyntaxClassification` to SyntaxBuilder

### PR DESCRIPTION
**Pull request by kimdv**

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

File: Sources/SwiftSyntaxBuilder/generated/ResultBuilders.swift [hunk 1]
```diff
@@ -264,6 +264,11 @@ public struct AttributeListBuilder {
   public static func buildExpression(_ expression: CustomAttribute) -> Self.Component {
     return buildExpression(.init (expression))
   }
+  /// If declared, provides contextual type information for statement
+  /// expressions to translate them into partial results.
+  public static func buildExpression(_ expression: IfConfigDecl) -> Self.Component {
+    return buildExpression(.init (expression))
+  }
```
[the remark this answers is not in the document]
Comment (kimdv) [reply Sources/SwiftSyntaxBuilder/generated/ResultBuilders.swift:271]: It's also added for me on the main: Running 

``` console
swift-syntax % python3 ./build-script.py generate-source-code --toolchain /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2022-10-27-a.xctoolchain/usr
** Generating gyb Files **
** Done Generating gyb Files **
** Running code generation **
Building for debugging...
[25/25] Linking generate-swiftbasicformat
Build complete! (4.45s)
Building for debugging...
[8/8] Linking generate-swiftsyntaxbuilder
Build complete! (1.59s)
```

File: Sources/IDEUtils/generated/SyntaxClassification.swift [hunk 2]
```diff
@@ -69,306 +70,308 @@ extension SyntaxClassification {
     // Separate checks for token nodes (most common checks) versus checks for layout nodes.
     if childKind == .token {
       switch (parentKind, indexInParent) {
-      case (.expressionSegment, 5):
```
Comment (kimdv) [inline Sources/IDEUtils/generated/SyntaxClassification.swift:72]: The reason for another order, is that we sort `SYNTAX_NODES`

File: CodeGeneration/Sources/SyntaxSupport/TokenSpec.swift.gyb [hunk 3]
```diff
@@ -224,6 +224,7 @@ public class MiscSpec: TokenSpec { }
 public let SYNTAX_TOKENS: [TokenSpec] = [
 % for token in SYNTAX_TOKENS:
 %   class_name = type(token).__name__
+%   classification = "classification: \"%s\"" % token.classification.name if token.classification.name not in ['None', 'Keyword', 'ObjectLiteral', 'PoundDirectiveKeyword'] or class_name == 'Misc' and token.classification.name != 'None' else None
```
Comment (kimdv) [inline CodeGeneration/Sources/SyntaxSupport/TokenSpec.swift.gyb:227]: This can be pulled out in a separate PR if needed.

File: Sources/IDEUtils/generated/SyntaxClassification.swift [hunk 4]
```diff
@@ -69,306 +70,308 @@ extension SyntaxClassification {
...
+    case .deinitKeyword: 
       return .keyword
-    case .enumKeyword:
+    case .enumKeyword: 
       return .keyword
-    case .extensionKeyword:
+    case .extensionKeyword: 
       return .keyword
-    case .funcKeyword:
+    case .funcKeyword:
```
Comment (kimdv) [inline Sources/IDEUtils/generated/SyntaxClassification.swift:134]: Nyt sure why the parser adds a whitespace after the `:` as there is no whitespace in the generator 
Might be bug @ahoppen ?

**Comment by kimdv** [comment]
@swift-ci please test

File: CodeGeneration/Sources/SyntaxSupport/Classification.swift.gyb [hunk 5]
```diff
@@ -34,6 +34,25 @@ public class SyntaxClassification {
   }
 }
 
+public class ChildClassification {
+  public let parent: Node
+  public let childIndex: Int
+  public let isToken: Bool
+  public let classification: SyntaxClassification?
+  public let force: Bool
+  public let swiftForce: String
```
Comment (ahoppen) [inline CodeGeneration/Sources/SyntaxSupport/Classification.swift.gyb:43]: If you want to have this property, I think it should be a computed property. But I would prefer to just do the conversion to string on the call side.

File: CodeGeneration/Sources/generate-swiftideutils/GenerateIDEUtils.swift [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+import SwiftSyntaxBuilder
+import Utils
+
+@main
+struct GenerateSwiftStaticFormat: ParsableCommand {
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftideutils/GenerateIDEUtils.swift:19]:
```suggestion
struct GenerateSwiftIDEUtils: ParsableCommand {
```

File: CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    leadingTrivia: .docLineComment(copyrightHeader),
+    attributes: AttributeList {
+      AttributeSyntax(
+        attributeName: TokenSyntax.identifier("_spi"),
+        leftParen: .leftParen,
+        argument: .identifier(.identifier("RawSyntax")),
+        rightParen: .rightParen)
+    },
+    path: [AccessPathComponent(name: "SwiftSyntax")]
+  )
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift:44]: What about

```suggestion
  ImportDecl("import @_spi(RawSyntax) import SwiftSyntax")
```

File: CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  
+  EnumDeclSyntax("public enum SyntaxClassification") {
+    for classification in SYNTAX_CLASSIFICATIONS {
+      EnumCaseDecl("/// \(classification.description)\ncase \(classification.swiftName)")
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift:48]: I think I would prefer a multi-line string literal here. Makes it easier to read IMO.

File: Sources/IDEUtils/generated/SyntaxClassification.swift (hunk 4 above)
Comment (ahoppen) [reply Sources/IDEUtils/generated/SyntaxClassification.swift:134]: I think formatting does because `:` is considered a punctuator and we want the whitespace in e.g. `[String: Int]`. I think it’s fine if we just keep the whitespace. It doesn’t really do any harm.

**Review by ahoppen — COMMENTED** [review]
Thanks 🙏

File: CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift (hunk 7 above)
Comment (kimdv) [reply CodeGeneration/Sources/generate-swiftideutils/SyntaxClassificationFile.swift:44]: I tried to do that, but it an error

```bash
SwiftSyntaxBuilder/Syntax+StringInterpolation.swift:179: Fatal error: Extraneous text in snippet: '@_spi(RawSyntax) import SwiftSyntax'
FAIL: Source generation using SwiftSyntaxBuilder failed
```

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]
Thanks. Looks good to me 👍

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@ahoppen is the CI on vacation ? 🍹

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci test

**Comment by kimdv** [comment]
@swift-ci please test

File: Sources/IDEUtils/CMakeLists.txt [hunk 9]
```diff
@@ -7,7 +7,7 @@
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
 add_library(IDEUtils STATIC
-  gyb_generated/SyntaxClassification.swift
```
Comment (kimdv) [inline Sources/IDEUtils/CMakeLists.txt:10]: @ahoppen I need to add this right?

**Comment by kimdv** [comment]
@swift-ci please test

File: Sources/IDEUtils/CMakeLists.txt (hunk 9 above)
Comment (ahoppen) [reply Sources/IDEUtils/CMakeLists.txt:10]: Yes, this looks correct to me.

**Comment by kimdv** [comment]
@ahoppen I get the error 
```
Only in /Users/kimdevos/repository/swift/swift-syntax/Sources/SwiftBasicFormat/generated: BasicFormat.swift
FAIL: Gyb-generated files committed to repository do not match generated ones. Please re-generate the gyb-files using the following command, open a PR to the SwiftSyntax project and merge it alongside the main PR.$ swift-syntax/build-script.py generate-source-code --toolchain /path/to/toolchain.xctoolchain/usr
```

but when I run it locally then I cannot see any changes?

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
@swift-ci please test
